backend/controllers/passController.js

```diff
@@ -13,27 +13,34 @@ const generatePass = asyncHandler(async (req, res) => {
 
     const user = await User.findOne({ _id: req.user._id })
     if (user) {
-      // console.log(user);
-      const newPass = await Pass.create({
-        name,
-        phone,
-        email,
-        duration,
-        start,
-        checkInStatus: true,
-        generateId: user.id,
-        userName: req.user.name,
-      })
-      if (newPass) {
-        res.status(201).json({
-          success: true,
-          newPass: newPass,
-          message: "Pass generated successfully",
+      if (user.role == "admin") {
+        // console.log(user);
+        const newPass = await Pass.create({
+          name,
+          phone,
+          email,
+          duration,
+          start,
+          checkInStatus: true,
+          generateId: user.id,
+          userName: req.user.name,
         })
+        if (newPass) {
+          res.status(201).json({
+            success: true,
+            newPass: newPass,
+            message: "Pass generated successfully",
+          })
+        }
+      } else {
+        throw new Error("You are not authorised to create Pass!")
       }
+    } else {
+      throw new Error("Signup to create Pass!")
     }
   } catch (err) {
     console.log(err)
+    res.status(400).json(err?.message)
```

Comment:
Add same logic in error handling middleware and use throw here also.

Reply:
done

Reply:
Can you try to signup with the following scenario

1. Run the server and wait till it says db connected
2. Turn off your internet connection
3. Send a signup request. 

Tell me what is showing in postman and in the terminal

Reply:
I think it should be internal server error but i will cross check from postman

Reply:
yes its the internal server error

Reply:
but message is not displaying properly as it should be. Need to look into it